DiscordIan/Module/OpenWeather.cs

```diff
@@ -0,0 +1,317 @@
+                        locale = address.City + ", " + address.State;
+                        string latlong = String.Format("{0},{1}", coordinates.Lat.ToString(), coordinates.Lng.ToString());
+
+                        await _cache.SetStringAsync(location, latlong,
+                            new DistributedCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(4) });
+
+                        await _cache.SetStringAsync(latlong, locale,
+                            new DistributedCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(4) });
+                    }
+                }
+                catch (Exception ex)
+                {
+                    //Geocode failed, revert to OpenWeatherMap's built-in resolve.
+                }
+            }
+            else
+            {
+                coordinates = new Model.Geocodio.Location() { Lat = double.Parse(coords.Split(",")[0]), Lng = double.Parse(coords.Split(",")[1]) };
+            }
+
+            string message;
+            Discord.Embed embed;
+
+            if (coordinates == null || locale == null)
+                (message, embed) = await GetWeatherResultAsync(location);
+            else
+                (message, embed) = await GetWeatherResultAsync(coordinates, locale);
+
+
+            if (message == null)
+            {
+                await ReplyAsync("No weather found, sorry!");
+            }
+            else
+            {
+                await ReplyAsync(message, false, embed);
+            }
+        }
+
+        private async Task<(string, Discord.Embed)>
+            GetWeatherResultAsync(Model.Geocodio.Location coordinates, string location)
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "User-Agent", "DiscorIan Discord bot" }
+            };
+
+            var uriCurr = new Uri(string.Format(_options.IanOpenWeatherMapEndpointCoords,
+                HttpUtility.UrlEncode($"{coordinates.Lat}"),
+                HttpUtility.UrlEncode($"{coordinates.Lng}"),
+                _options.IanOpenWeatherKey));
+
+            var uriFore = new Uri(string.Format(_options.IanOpenWeatherMapEndpointForecast,
+                HttpUtility.UrlEncode($"{coordinates.Lat}"),
+                HttpUtility.UrlEncode($"{coordinates.Lng}"),
+                _options.IanOpenWeatherKey));
+
+            var responseCurr = await _fetchXMLService
+                .GetAsync<WeatherCurrent.Current>(uriCurr, headers);
+
+            if (responseCurr.IsSuccessful)
+            {
+                string message = string.Empty;
+                var data = responseCurr.Data;
+
+                var responseFore = await _fetchJsonService
+                    .GetAsync<WeatherForecast.Forecast>(uriFore, headers);
+
+                if (responseFore.IsSuccessful)
+                {
+                    var foredata = responseFore.Data;
+                    var today = foredata.Daily[0];
```

Comment:
It doesn't look like you're checking if `foredata.Daily` is null or has elements before deference.